src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -644,6 +655,10 @@ protected virtual Solution AdjustReloadedSolution(Solution oldSolution, Solution
                 {
                     newSolution = newSolution.WithDocumentText(docId, this.GetOpenDocumentText(oldSolution, docId), PreservationMode.PreserveIdentity);
                 }
+                else if (newSolution.ContainsAdditionalDocument(docId))
```

Comment:
Honestly this is giving me hope making GetOpenDocumentIds do the right thing is actually the right way to go: your audit turned up a legitimate bug. :smile: